core/src/main/java/com/epam/esm/jpa/specification/CertificateSpecification.java

```diff
@@ -0,0 +1,88 @@
+package com.epam.esm.jpa.specification;
+
+import com.epam.esm.model.dto.search.GiftSearchDto;
+import com.epam.esm.model.entity.GiftCertificateEntity;
+import com.epam.esm.util.DaoConstants;
+import com.epam.esm.util.SearchConstants;
+import org.springframework.data.jpa.domain.Specification;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.util.List;
+
+public class CertificateSpecification {
+    public static Specification<GiftCertificateEntity> bySearchRequest(GiftSearchDto giftSearchDto) {
+        String giftNamePrefix = giftSearchDto.getNamePrefix();
+        String giftDescriptionPrefix = giftSearchDto.getDescriptionPrefix();
+        List<String> tagNamePrefixes = giftSearchDto.getTagNamePrefixes();
+
+        Specification<GiftCertificateEntity> finalSpec = (Specification<GiftCertificateEntity>) (root, query, criteriaBuilder) -> null;
+
+        if (giftNamePrefix != null) {
+            finalSpec = Specification
+                    .where(finalSpec
+                            .and(giftNameLike(giftNamePrefix, giftSearchDto.getSortField(), giftSearchDto.getSortMethod())));
+        }
+        if (giftDescriptionPrefix != null) {
+            finalSpec = Specification
+                    .where(finalSpec
+                            .and(giftDescriptionLike(giftDescriptionPrefix, giftSearchDto.getSortField(), giftSearchDto.getSortMethod())));
+        }
+        if (tagNamePrefixes != null) {
+            finalSpec = Specification
+                    .where(finalSpec
+                            .and(giftTagNamePrefixesLike(tagNamePrefixes, giftSearchDto.getSortField(), giftSearchDto.getSortMethod())));
+        }
+
+        return finalSpec;
+    }
+
+    private static Specification<GiftCertificateEntity> giftNameLike(String giftNamePrefix, String sortField, String sortMethod) {
+        return (root, query, criteriaBuilder) -> {
+            defineSortMethodAndColumn(sortField, sortMethod, root, query, criteriaBuilder);
+
+            return criteriaBuilder.like(root.get(DaoConstants.GIFT_FIELD_NAME),
+                    DaoConstants.ZERO_OR_MORE_ELEMENTS_WILDCARD + giftNamePrefix + DaoConstants.ZERO_OR_MORE_ELEMENTS_WILDCARD);
```

Comment:
the name of your field is giftNamePrefix so I would expect that the wildcard should be only at the end of like condition